### ISSUE TITLE
PyTorch is super bloated, even the data loader has too many lines of code.

### ISSUE THREAD

**Issue by power9799** [pytorch/pytorch#114888]
### 🐛 Describe the bug

Hi, why is PyTorch so bloated and slow? I don't get it, why do you need so many lines of code just to load a custom dataset. The extra features like transform can an option feature separate from data loader.  Someone shouldn't need  around 500 lines of code to load a dataset.... that is on the UI. If someone decompiles the source code, it is probably around 10,000 lines.
Can't someone optimize at least the data loader?

### Versions

python 3.11 , PyTorch nightly

**Comment by ethicalhacker7192** [comment]
I agree, PyTorch is too bloated, it got rid of the support for Tesla k80s and I literally had to build python3.10 from source code and download their package manually, in my version all you needed to do was torch.load() and that was it, now we need like 4 lines just to load our optimizers and training data and weights, can you all go back to the way we used to have it? It was much better back then, also I disagree with the way you all get rid of support for every GPU eventually, I would love to use my Tesla K80 with PyTorch 2.1 but that is just a fantasy apparently.

**Comment by power9799** [comment]
> I agree, PyTorch is too bloated, it got rid of the support for Tesla k80s and I literally had to build python3.10 from source code and download their package manually, in my version all you needed to do was torch.load() and that was it, now we need like 4,000 lines just to load our optimizers and training data and weights, can you all go back to the way we used to have it? It was much better back then, also I disagree with the way you all get rid of support for every GPU eventually, I would love to use my Tesla K80 with PyTorch 2.1 but that is just a fantasy apparently.

They can just make people import an extra library for the old gpus and cpus.

**Comment by malfet** [comment]
This feels like a great topic to raise on https://discuss.pytorch.org/
Closing, because as bug report it does not have enough details on what is not work or on designed outcome. 
But please do not hesitate if you run into an issue with PyTorch or want to propose a new feature (faster dataloader sounds like an interesting feature, but then it would be useful if one can outline what is wrong with the current one)
